docs/ref/models/conditional-expressions.txt

```diff
@@ -60,6 +62,9 @@ Some examples::
     >>> # Complex conditions can be created using Q objects
     >>> When(Q(name__startswith="John") | Q(name__startswith="Paul"),
     ...      then='name')
+    >>> # Other expressions that will evaluate to a boolean value
+    >>> from django.db.models.expressions import Exists
+    >>> When(Exists(Client.objects.filter(registered__on=datetime.date.today())))
```

Comment:
Might want to use a `F` or an `OuterRef` instead of a fixed value as this expression wouldn't make much sense.